### PR TITLE
fix float reward, add stake verification

### PR DESCRIPTION
**Pull request by bogdan34**

### CODE REVIEW

File: core/service/transaction/stake.ts [hunk 1]
```diff
@@ -94,6 +97,11 @@ class TransactionStakeService implements IAssetService<IAssetStake> {
         if (!airdropCheck.success) {
             errors = airdropCheck.errors;
         }
+
+        if ((trs.asset.amount < MIN_STAKE_AMOUNT || trs.asset.amount % 1 !== 0) && FailService.isVerifyStake(trs)) {
```
Comment (axelsheva) [inline core/service/transaction/stake.ts]: trs.asset.amount % 100000000 !== 0, and 100000000 must be const

File: core/service/fail.ts [hunk 2]
```diff
@@ -32,6 +35,16 @@ class FailService {
             && BlockStorageService.getHeaders(trs.blockId).height >
             config.CONSTANTS.START_FEATURE_BLOCK.FIX_TRANSACTION_IN_THE_PAST_BLOCK_HEIGHT;
     }
+
+    public stakeReward(trs: TransactionModel<IAssetVote>): boolean {
+        return failedVoteTransaction.has(trs.id)
+    }
+
+
+    public isVerifyStake(trs: TransactionModel<IAssetStake>): boolean {
```
Comment (Verbalist) [inline core/service/fail.ts]: isVerifyStakeMinAmount

File: core/service/fail.ts [hunk 3]
```diff
@@ -32,6 +35,16 @@ class FailService {
             && BlockStorageService.getHeaders(trs.blockId).height >
             config.CONSTANTS.START_FEATURE_BLOCK.FIX_TRANSACTION_IN_THE_PAST_BLOCK_HEIGHT;
     }
+
+    public stakeReward(trs: TransactionModel<IAssetVote>): boolean {
```
Comment (Verbalist) [inline core/service/fail.ts]: stakeFloatReward

File: core/service/fail.ts [hunk 4]
```diff
@@ -1,10 +1,13 @@
 import { BlockId } from 'shared/repository/block';
 import config from 'shared/config';
-import { Transaction } from 'shared/model/transaction';
+import { IAssetStake, IAssetVote, Transaction, TransactionModel } from 'shared/model/transaction';
 import BlockStorageService from 'core/service/blockStorage';
 
 const FAIL_ROUND_BUILD_BLOCK_ID_245191 = '9795e3314bc7a550f084face0fe40e4feaa7d1fe8d45545b2b19f894076ec328';
 const TRUE_ROUND_BUILD_BLOCK_ID_245191 = '369538bb46848894a66a7bc974d9c3ea76e9e7e7d5137a853a88db6f663a0fb5';
+const failedVoteTransaction = new Set([
```
Comment (axelsheva) [inline core/service/fail.ts]: `FAILED_VOTE_TRANSACTIONS`

File: core/service/fail.ts [hunk 5]
```diff
@@ -32,6 +35,16 @@ class FailService {
             && BlockStorageService.getHeaders(trs.blockId).height >
             config.CONSTANTS.START_FEATURE_BLOCK.FIX_TRANSACTION_IN_THE_PAST_BLOCK_HEIGHT;
     }
+
+    public stakeFloatReward(trs: TransactionModel<IAssetVote>): boolean {
```
Comment (axelsheva) [inline core/service/fail.ts]: isFailedVoteReward

File: core/service/fail.ts [hunk 6]
```diff
@@ -32,6 +35,16 @@ class FailService {
...
+        return failedVoteTransaction.has(trs.id);
+    }
+
+
+    public isVerifyStakeMinAmount(): boolean {
```
Comment (axelsheva) [inline core/service/fail.ts]: Should not be here

Comment (bogdan34) [reply core/service/fail.ts]: OK=)

**Pull request merged**